rust/lance/src/dataset.rs

```diff
@@ -2383,6 +2383,20 @@ mod tests {
         .unwrap();
 
         assert_eq!(actual, expected);
+
+        // Validate we can still read after re-instantiating dataset, which
+        // clears the cache.
+        let dataset = Dataset::open(test_uri).await.unwrap();
```

Comment:
Is this where the bug was reproduced?

Reply:
Yes, once we reloaded the table, the in-memory version of the PageTable was gone and we are reading from disk, which is where the problems start.